### PR TITLE
f-checkout@0.85.0 Exchange OTAC to JWT

### PR DESCRIPTION
**Pull request by pkedziora**
`otacToAuthExchanger` prop that provides functionality to exchange OTAC to JWT token for guest checkout. The function will be implemented outside of f-checkout and passed as prop as it's not f-checkout responsibility to retrieve authentication details.

When OTAC is exchanged to JWT, the isLoggedIn flag stays `false` to indicate that we are still in 'guest' mode, while authToken gets set to correct value.

---

## UI Review Checks
No UI changes

- [ ] README and/or UI Documentation has been [created|updated]
- [x] Unit tests have been [created|updated]
- [ ] This code has been checked with regard to [our accessibility standards](http://fozzie.just-eat.com/documentation/general/accessibility/checklist)

## Browsers Tested

- [x] Chrome (latest)
- [ ] Internet Explorer 11
- [ ] Mobile (Please list device/browser – Ideally one iPhone model and one Android)

### List any other browsers that this PR has been tested in:

- [View the Browser Support Checklist](http://fozzie.just-eat.com/documentation/general/browser-support)

### CODE REVIEW

File: packages/components/organisms/f-checkout/src/components/Checkout.vue [hunk 1]
```diff
@@ -215,6 +215,11 @@ export default {
             default: ''
         },
 
+        otacToAuthExchanger: {
+            type: Function,
+            default: () => ''
```
Comment (tommcclean) [inline packages/components/organisms/f-checkout/src/components/Checkout.vue]: Suggestion: Could log a not implemented message

**Comment by tommcclean** [comment]
Remember to update the props list in the readme

File: packages/components/organisms/f-checkout/src/components/Checkout.vue (hunk 1 above)
Comment (pkedziora) [reply packages/components/organisms/f-checkout/src/components/Checkout.vue]: Logger would not be available here as per: https://vuejs.org/v2/guide/components-props.html
"Note that props are validated before a component instance is created, so instance properties (e.g. data, computed, etc) will not be available inside default or validator functions."

**Comment by pkedziora** [comment]
@tommcclean Readme updated

File: packages/components/organisms/f-checkout/src/components/Checkout.vue (hunk 1 above)
Comment (pkedziora) [reply packages/components/organisms/f-checkout/src/components/Checkout.vue]: @tommcclean I replaced it with default implementation that throws not implemented exception.

**Review by tommcclean — APPROVED** [review]

**Review by simonsandersje — APPROVED** [review]

**Pull request merged**